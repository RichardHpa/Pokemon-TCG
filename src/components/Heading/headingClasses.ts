export const headingClasses = {
  base: 'font-bold tracking-tight text-gray-900 dark:text-white mb-2',
  level: {
    1: 'text-5xl font-extrabold',
    2: 'text-4xl',
    3: 'text-3xl',
    4: 'text-2xl',
    5: 'text-xl',
    6: 'text-lg',
  },
}
