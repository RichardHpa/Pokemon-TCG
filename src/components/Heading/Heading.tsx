import { createElement } from 'react'
import clsx from 'clsx'

import { headingClasses } from './headingClasses'

import type { ReactNode } from 'react'

export type HeadingProps = {
  className?: string
  level?: '1' | '2' | '3' | '4' | '5' | '6'
  children: ReactNode
}

export const Heading = ({ children, level = '1' }: HeadingProps) => {
  return createElement(
    `h${level}`,
    { className: clsx(headingClasses.base, headingClasses.level[level]) },
    children,
  )
}
