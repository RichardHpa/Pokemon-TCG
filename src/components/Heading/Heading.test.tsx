import { render, screen } from '@testing-library/react'

import { Heading } from './Heading'

describe('Heading', () => {
  test('renders Heading', () => {
    const { container } = render(<Heading>Heading Text</Heading>)
    expect(screen.getByRole('heading', { level: 1, name: 'Heading Text' })).toBeInTheDocument()

    expect(container).toMatchSnapshot()
  })

  test.each(['1', '2', '3', '4', '5', '6'] as const)('renders Heading with level:%s', (level) => {
    render(<Heading level={level}>Heading Text</Heading>)
    expect(
      screen.getByRole('heading', { level: Number(level), name: 'Heading Text' }),
    ).toBeInTheDocument()
  })
})
