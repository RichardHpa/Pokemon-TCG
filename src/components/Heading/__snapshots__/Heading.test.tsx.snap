// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Heading renders Heading 1`] = `
<div>
  <h1
    class="font-bold tracking-tight text-gray-900 dark:text-white mb-2 text-5xl font-extrabold"
  >
    Heading Text
  </h1>
</div>
`;
